src/pages/ProductPage/ProductPage.tsx

```diff
@@ -1,44 +1,93 @@
+import { LoaderFunction } from "react-router-dom";
+import { useState } from "react";
+
 import Grid from "@mui/material/Unstable_Grid2";
 import { FavoriteBorder } from "@mui/icons-material";
 import Button from "@mui/material/Button";
 import CartCounter from "../../components/CartCounter/CartCounter";
 import { useLoaderData } from "react-router";
 import { getProduct } from "../service";
+import { useCart } from "../CartPage/CartContext";
 
 import styles from "./ProductPage.module.scss";
 
 import { Product } from "../types";
-import { LoaderFunction } from "react-router-dom";
 
 export const loader: LoaderFunction = async ({ params }) => {
   if (!params.productId) throw new Error();
   return await Promise.resolve(getProduct(params.productId as string));
 };
 
 const ProductPage = () => {
-  const product = useLoaderData() as Product;
+  const { id, image, title, price, description } = useLoaderData() as Product;
+
+  const { cart, addItem, increaseAmount, decreaseAmount } = useCart();
+  const [initialAmount, setInitialAmount] = useState(1);
+  const isProductInCart = cart.some((product) => product.id === id);
+  const productInCart = cart.find((product) => product.id === id);
```

Comment:
Always think if it's possible to reduce number of array itterations.
Here you can do the following
```suggestion
  const productInCart = cart.find((product) => product.id === id);
  const isProductInCart = !!productInCart;
```